### PR TITLE
Docs/async blog

### PR DESCRIPTION
**Pull request by JP-Ellis**
## :memo: Summary

Adding a new blog post to highlight the contributions from @valkolovos that made it in v2.2.1 of Pact Python 🚀

### CODE REVIEW

File: docs/blog/posts/2024/07-26 asynchronous message support.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    .set_source("/path/to/pacts")
+    .set_state(provider.state_url)  # (2)
+    .add_transport(  # (3)
+        protocol="message",
+        path="/_pact/message",
+    )
+ )
+```
+
+1.  The provider URL is required but only used if generating both HTTP and message interactions.
```
Comment (YOU54F) [inline docs/blog/posts/2024/07-26 asynchronous message support.md:138]: maybe reword if the pacts to be verified contain both HTTP and message interactions.

generating interactions to me would be the consumer side

File: docs/blog/posts/2024/07-26 asynchronous message support.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+2.  The provider URL is required, though would only be used if the provider generates both HTTP and message interactions.
```
Comment (YOU54F) [inline docs/blog/posts/2024/07-26 asynchronous message support.md:139]: Should this say provider state url?

File: docs/blog/posts/2024/07-26 asynchronous message support.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+3.  This path is used by Pact to ensure that the provider is in the correct state before sending the message.
+
+Those familiar with HTTP interactions will notice that the process is very similar, with the key difference of the additional `add_transport` method. This configure a HTTP endpoint on the provider that Pact will call to trigger the provider to send the message.
```
Comment (YOU54F) [inline docs/blog/posts/2024/07-26 asynchronous message support.md:142]: configures

File: docs/blog/posts/2024/07-26 asynchronous message support.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    """
+    A simple HTTP provider that sends messages to the consumer.
+
+    This would typically use the same underlying functions that would generate messages, except that instead of being sent into the message queue, they are sent to the consumer's HTTP server.
+    """
+
+provider = Provider()
+
+(
+    Verifier()
```
Comment (YOU54F) [inline docs/blog/posts/2024/07-26 asynchronous message support.md:127]: Is it worth mentioning that this is a departure from the existing interfaces `MessageProvider`

File: docs/blog/posts/2024/07-26 asynchronous message support.md (hunk 3 above)
Comment (YOU54F) [reply docs/blog/posts/2024/07-26 asynchronous message support.md:142]: >1.  The provider URL is required but only used if generating both HTTP and message interactions.
> 2.  The provider URL is required, though would only be used if the provider generates both HTTP and message interactions.

are the provider.url and provider.state_url only used for http, will they be ignored for message interactions?

mentioned in points 1/2 above.

for 3, `path="/_pact/message"`

> This configure a HTTP endpoint on the provider

If we are validating a message interaction only, no http interactions. I assume we need 1. the provider url to prefix this path

File: docs/blog/posts/2024/07-26 asynchronous message support.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+Asynchronous messages play a crucial role in building resilient and scalable systems. They allow services to communicate with each other without blocking, which can be particularly useful when the sender and receiver are not always available at the same time. However, verifying these interactions is challenging due to the wide variety of messaging systems and protocols.
+
+Pact simplifies this process by focusing on the content of the messages rather than their transport mechanisms. This approach allows defining expected message exchanges and verifying their adherence independently of messaging systems and protocols. The Pact specification provides a way to verify these interactions, but until now, Pact Python has not supported this feature.
```
Comment (YOU54F) [inline docs/blog/posts/2024/07-26 asynchronous message support.md:14]: Worth linking out to the main site page on message pact?

https://docs.pact.io/getting_started/how_pact_works#non-http-testing-message-pact

> The Pact specification provides a way to verify these interactions, but until now, Pact Python has not supported this feature.

Technically Pact python has (at least for V3 Pacts), using https://github.com/pact-foundation/pact-message-ruby but the implementation is incomplete, and has a couple of bugs

**Comment by YOU54F** [comment]
Few minor comments, but looks and reads great. nice example.

Great work @valkolovos!

**Review by YOU54F — APPROVED** [review]
Feel free to cherry pick any amendments from my comments, otherwise all good with me

File: docs/blog/posts/2024/07-26 asynchronous message support.md (hunk 5 above)
Comment (JP-Ellis) [reply docs/blog/posts/2024/07-26 asynchronous message support.md:14]: Link added, thanks :)

File: docs/blog/posts/2024/07-26 asynchronous message support.md (hunk 1 above)
Comment (JP-Ellis) [reply docs/blog/posts/2024/07-26 asynchronous message support.md:138]: Reworded. I was thinking of generation in terms of Pact generating interactions (as in, mocking the other end), but that is clearly confusing.

File: docs/blog/posts/2024/07-26 asynchronous message support.md (hunk 2 above)
Comment (JP-Ellis) [reply docs/blog/posts/2024/07-26 asynchronous message support.md:139]: Fixed

File: docs/blog/posts/2024/07-26 asynchronous message support.md (hunk 3 above)
Comment (JP-Ellis) [reply docs/blog/posts/2024/07-26 asynchronous message support.md:142]: For your first part: I've disambiguated the state_url portion to make it clear it is for _both_.

For your second part: You have to define a `messages` transport at home point, though it technically could be done at the first. I'll rework the example to highlight this.

File: docs/blog/posts/2024/05-02 integrating rust ffi with pact python.md [hunk 6]
```diff
@@ -17,14 +17,14 @@ In this blog post, I will delve into how this is all achieved. From explaining h
 
 ## Briding Python and Binary Libraries
```
Comment (mefellows) [inline docs/blog/posts/2024/05-02 integrating rust ffi with pact python.md:18]: Should this be "bridging"?

File: docs/blog/posts/2024/07-26 asynchronous message support.md (hunk 4 above)
Comment (JP-Ellis) [reply docs/blog/posts/2024/07-26 asynchronous message support.md:127]: I personally want the blog post to just focus on what's coming, as opposed to how things were. 

I would like to create a more thorough upgrade guide once we are much closer to `v3` though.

**Review by mefellows — APPROVED** [review]
looks good. The only comment I have is to ensure the endpoint to verify messages is clear (i.e. making it clear what the API contract is for that endpoint to work). That need not be here

File: docs/blog/posts/2024/05-02 integrating rust ffi with pact python.md (hunk 6 above)
Comment (JP-Ellis) [reply docs/blog/posts/2024/05-02 integrating rust ffi with pact python.md]: Fixed

**Comment by JP-Ellis** [comment]
I've fixed a few issues raised by @YOU54F, and if I understand your comment @mefellows, I'll also add example payloads for the intermediate HTTP relay as I think that will make things clear.
